src/hotspot/share/gc/shenandoah/shenandoahHeap.cpp

```diff
@@ -1729,6 +1725,12 @@ void ShenandoahHeap::op_final_mark() {
       _free_set->rebuild();
     }
 
+    if (!collection_set()->is_empty()) {
+      if (ShenandoahVerify) {
+        verifier()->verify_before_evacuation();
+      }
+    }
+
```

Comment:
Why this move?

Reply:
That seems to be a leftover from when the verifier was not yet working correctly. I think it can be reverted.